app/perfunwind.cpp

```diff
@@ -441,10 +441,16 @@ void PerfUnwind::resolveCallchain()
     bool addedUserFrames = false;
     PerfSymbolTable *symbols = symbolTable(m_currentUnwind.sample->pid());
 
-    auto reportIp = [&](quint64 ip) -> bool {
+    auto reportIp = [&](quint64 ip, bool branchStack) -> bool {
         symbols->attachDwfl(&m_currentUnwind);
-        m_currentUnwind.frames.append(symbols->lookupFrame(ip, isKernel,
-                                            &m_currentUnwind.isInterworking));
+        int frame = symbols->lookupFrame(ip, isKernel, &m_currentUnwind.isInterworking);
+        // There are some required frames are not looked up for LBR unwinding method.
+        // Entered disasmFrames to accumulate frames for LBR to compute then Disassembly events costs within function/method.
+        if (branchStack) {
```

Comment:
as I mentioned in the past - I do not understand why we have to distinguish between disasm frames and non-disasm frames. a frame is a frame, either it applies to all views or none.

please expand your commit message with an example that shows why this is needed.